pkg/admission/validator/shoot.go

```diff
@@ -159,6 +159,17 @@ func (s *shoot) validateCreate(ctx context.Context, shoot *core.Shoot) error {
 		return err
 	}
 
+	// TODO: This check won't be needed after generic support to scale from zero is introduced in CA
+	// Ongoing issue - https://github.com/gardener/autoscaler/issues/27
+	workers := shoot.Spec.Provider.Workers
+	for i, worker := range workers {
+		workerFldPath := workersPath.Index(i)
+		if worker.Maximum != 0 && worker.Minimum < int32(len(worker.Zones)) {
```

Comment:
Why is the `worker.Maximum != 0` combined with the `worker.Minimum < int32(len(worker.Zones))` check?

I mean `worker.Maximum` needs always be greater than 0 (until we have scale to zero for gcp) and greater equal `worker.Minimum`, right? So should `worker.Maximum == 0` not be an independent check?

Reply:
You are right, @dkistner. The `worker.maximum` check need not be here. There is already a validation present [here](https://github.com/gardener/gardener-extension-provider-gcp/blob/21db727e7c36db6a5cf549e789c59c46fbb61683/pkg/apis/gcp/validation/shoot.go#L53-L55) for the `worker.Maximum` value. I shall update the code.